ambry-router/src/main/java/com/github/ambry/router/PutOperation.java

```diff
@@ -647,23 +652,32 @@ void fillChunks() {
         }
       }
       if (chunkFillingCompletedSuccessfully) {
-        // If the blob size is less then 4MB or the last chunk size is less than 4MB, than this lastChunk will be
+        // If the blob size is less than 4MB or the last chunk size is less than 4MB, than this lastChunk will be
```

Comment:
the logic in this if statement looks good to me. However, I do think the comments should be more detailed. There are several case we have to deal with within this if statement and there are two decisions we need to make in those cases. 
Decision 1: Is this a simple blob?
Decision 2:  Do we have to send the last chunk to ambry servers?

Case 1: there is no data at all.
In this case: last chunk is not null, and is building, chunkCounter == 0  so the isSimpleBlob == options.isChunkUpload. calling mayResolveAwaitingChunk is an nop since there is no chunk awaiting. And we will send this 0 byte chunk to ambry server at line 667.

Case 2:  there are exactly 4MB data, and there is no empty bytebuf after.
In this case: last chunk is null, chunkCounter == 0. isSimpleBlob = isChunkUpload. maybeResolveAwaitingChunk would send the data to ambry server. Since the lastChunk == null, it will skip the next two if statements.

Case 3: there are exactly 4MB data, but there is an empty bytebuf after 4MB.
In this case: last chunk is not null, and is building with readableBytes == 0. chunkcounter == 1 so the isSimpleBlob = isChunkUpload. maybeResolveAwaitingChunk would send the first chunk out to ambry server.

Case 4: there are more than 4MB data. 
In this case: isSimpleBlob would be false. and it depends on if we have any data in the last chunk to determine weather we want to send it to server.

Reply:
Yes, Justin's comment will be helpful.

Reply:
Updated the code, so we don't need these comments here.
I have added the comments elsewhere, where `isSimpleBlob` status is being set.